keps/sig-network/4974-deprecate-endpoints/README.md

```diff
@@ -0,0 +1,612 @@
+would only be changed when the administrator chose to disable the
+controllers, which could happen at any time.
+
+### Version Skew Strategy
+
+The only non-opt-in behavioral change is fixing the apiserver
+aggregation and proxying APIs to use EndpointSlices rather than
+Endpoints internally, which does not present any skew issues (since
+EndpointSlices have already existed for a long time at this point).
+
+(When the conformance criteria change to allow disabling the Endpoints
+controller, this would only apply to clusters that are _fully_ at the
+new version, to avoid skew issues. More specifically: it is not
+conforming to disable the Endpoints controller if you still have any
+apiservers that use Endpoints for apiserver aggregation and proxying.)
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+The KEP does not define a new feature, it just proposes that we
+document that users are allowed to disable the Endpoints and
+EndpointSlice Mirroring controllers.
+
+###### Does enabling the feature change any default behavior?
+
+The KEP does not define a new feature that can be enabled.
+
+(Obviously disabling the endpoints controllers changes default
+behavior, but this would be because the administrator chose to do
+that, not something that would happen automatically.)
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+The KEP does not define a new feature that can be enabled/disabled.
+
+(If an administrator chooses to disable the endpoints controllers, and
+then decides this was a bad idea, they can re-enable them, and even if
+something had previously deleted all of the autogenerated Endpoints
+objects, re-enabling the endpoints controller would regenerate them.)
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+N/A
+
+###### Are there any tests for feature enablement/disablement?
+
+N/A: The KEP does not define a new feature that can be enabled/disabled.
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+Disabling the endpoints controllers in a cluster where some
+third-party components depend on them could have arbitrarily bad
+effects.
+
+###### What specific metrics should inform a rollback?
+
+We may add metrics monitoring usage of the v1.Endpoints API, but
+ideally you would have looked at those metrics _before_ disabling the
+Endpoints controller.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+N/A
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+N/A
+
+### Monitoring Requirements
+
+###### How can an operator determine if the feature is in use by workloads?
+
+The feature would be enabled by the operator.
```

Comment:
I don't see how this is true. We are deprecating endpoints in our code base. I did not read that the operator would trigger deprecation.

Reply:
Ah, the PRR section was originally written from the viewpoint of "the feature" == "disabling the controllers", and an earlier reviewer pointed out that this was wrong and I rewrote it but I guess I missed some spots.